routes/posts.ts

```diff
@@ -0,0 +1,86 @@
+import express, { Response, Request } from "express";
+import mongoose from "mongoose";
+const router = express.Router();
+const Post = require("../models/posts");
+
+router.post("/", (req: Request, res: Response) => {
+    Post.find({ image: req.body.image }).exec().then((posts) => {
+        if (posts.length > 0) {
+            return res.status(409).json({ message: "That image was already posted" });
+        } else {
+            const post = new Post({
+                _id: new mongoose.Types.ObjectId(),
+                title: req.body.title,
+                tags: req.body.tags,
+                image: req.body.image,
+                author: req.body.author, //autorIdRead() from localstorage will be used here
+                likes: 0,
```

Comment:
Tutaj Post ma trochę za dużo odpowiedzialności. Dodajecie do niego likes i comments. Warto pomyśleć o zasadzie Open-Closed czy SRP. I np. zmienić zależność w drugą stronę. Czy do istnienia feature, jakim są posty potrzebne są koniecznie lajki czy komentarze? Może to zapewne istnieć niezależnie. Tzn. komentarze mogłyby być zupełnie osobnym modułem - czy przynajmniej dokumentem w bazie danych. 

Takie dodawanie ciągle do jednego bytu potem powoduje kolekcje, których dokumenty mają po 200 pól. Oczywiście tutaj projekt jest mniejszy, ale warto myśleć o tym jak zrobić kawałek kodu, którego już nie będzie trzeba zmieniać, a będzie umożliwiał dodawanie nowych funkcjonalności.